eslint/README.md

```diff
@@ -0,0 +1,88 @@
+# [ESL](../../../) ESL migration support
+
+Version: *4.12.0*
+
+Authors: *Natalia Smirnova, Alexey Stsefanovich (ala'n)*.
+
+<a name="intro"></a>
+
+This article aims to assist the smooth migration process from older versions of ESL to the latest version. To support this transition, we have developed an ESLint plugin designed specifically for ESL library. This plugin targets deprecated features and aliases within the library, offering a seamless means of identifying these outdated elements. Additionally, it suggests suitable replacements, thereby ensuring a hassle-free upgrade to the newest version of ESL.
+
+### Installation
+To use custom ESLint plugin, you need to install it as a development dependency in your project. You can do this using npm:
+
+```bash
+npm install --save-dev @exadel/eslint-plugin-esl
+```
+
+### Usage
+To configure plugins inside of a configuration file, use the plugins key, which contains a list of plugin names:
+
+```json
+{
+  // ...
+  "plugins": [
+    "@exadel/esl"
+  ]
+  // ...
+}
+```
+
+Or in YAML:
+```js
+  plugins: 
+    - "@exadel/esl"
+```
+
+Once you have added the plugin to your project, you can define specific ESLint rules for handling ESL deprecations in your ESLint configuration file. Here are some examples of how to configure them:
+
+```json
+{
+  // ...
+  "rules": {
+    // Rule for deprecated `generateUId` alias for randUID
+    "@exadel/esl/deprecated-4/generate-uid" : "warn", // warn | error
+
+    // Rule for deprecated `deepCompare` alias for isEqual
+    "@exadel/esl/deprecated-4/deep-compare" : "warn", // warn | error
+
+    // Rule for deprecated `EventUtils` alias for ESLEventUtils
+    "@exadel/esl/deprecated-4/event-utils" : "warn", // warn | error
+
+    // Rule for deprecated `TraversingQuery` alias for ESLTraversingQuery
+    "@exadel/esl/deprecated-4/traversing-query" : "warn", // warn | error
+
+    // Rule for deprecated `ToggleableActionParams` alias for ESLToggleableActionParams
+    "@exadel/esl/deprecated-4/toggleable-action-params" : "warn" // warn | error
+  }
+  // ...
+}
```

Comment:
We could desscribe it in the separate rules section, grouping by stage (4 / 5, we have eno 5 stage rules for now) and provide detailed description for what does the rule deprecates, so it will also describes the list of breaking changes that mentioned in stages/presets